src/mutations/Metadata.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of ciboard
+
+ * Copyright (c) 2021, 2022 Andrei Stepanov <astepano@redhat.com>
+ * 
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { gql } from '@apollo/client';
+
+export const MetadataUpdate = gql`
+    mutation MetadataUpdate(
+        $_id: ID
+        $testcase_name: String
+        $testcase_name_is_regex: Boolean
+        $priority: Int
+        $product_version: String
+        $payload: JSON
+    ) {
+        metadata_update(
+            _id: $_id
+            testcase_name: $testcase_name
+            testcase_name_is_regex: $testcase_name_is_regex
+            priority: $priority
+            product_version: $product_version
+            payload: $payload
+        ) {
```

Comment:
I do not know what to reply here.
Is this blocker for merging this PR?

Reply:
@mgrabovsky where is anti-pattern?
GraphQL - is a way to doccument the arguments.
With the "blog" approach you lose this ability.

Take a look https://api.atlassian.com/graphql  (I think atlassian is enought authoritative platform).

Or any other https://graphql.org/users/ 

![image](https://user-images.githubusercontent.com/12443962/189001481-1bc2eeba-d18f-44fa-bfe1-c2d965de98cb.png)

Reply:
I'm sorry if you misunderstood my comment. My intent was to be helpful and constructive in suggesting a more maintainable and modern approach to GraphQL schema design.

I don't understand your comment “GraphQL - is a way to doccument the arguments. With the "blog" approach you lose this ability.” You can document fields in an input type the same way you document arguments to a query or a mutation.

Regarding your appeal to authority: One, I don't consider Atlassian to be the epitome of good design (see Jira, for instance). And two, even if they were, you cherry-picked one of only two mutations in their API that use this form. All of the rest use the input-and-payload type form:
![Screenshot from 2022-09-08 13-37-08](https://user-images.githubusercontent.com/134321/189112680-cdbf867e-fe95-4849-ada1-5f03c8cf7295.png)

The link l posted is from the Apollo GraphQL blog. We're using Apollo GraphQL. If you trust Apollo with their GraphQL library, you could trust them with their design approach as well. [GitHub](https://docs.github.com/en/graphql/reference/mutations) uses this approach as well. [Here's a talk](https://www.youtube.com/watch?v=pJamhW2xPYw) on GraphQL schema design from a person at GitHub where he goes into more depth why it's useful to structure the API in that way.

The [GraphQL docs](https://graphql.org/learn/schema/#input-types) on input fields mention a nice metaphor which applies in this case as well: “This is particularly valuable in the case of mutations, where you might want to pass in a whole object to be created.”

Reply:
@mgrabovsky you mix concepts.

The full quote is:

```
 But you can also easily pass complex objects. This is particularly valuable in the case of mutations, where you might want to pass in a whole object to be created. 
```
I do pass complex object: example: `payload: JSON`
When it is approriate. This perfectly covers this quote.

You recommend to wrap ALL variables into Object. I say this is wrong. Since you are missing description, and clearity that influence on backend logic.